### PR TITLE
#149 - add emission mode to grammar

### LINKED ISSUE #149: Add grammar statement for setting the emission mode (batch / streaming)

### PR DESCRIPTION
**Pull request by Knostpe**
Introduces an emission mode in the grammar in form of the Keyword EMISSION MODE and a literal. The literal should either be  "default batching", "safe batching" or "streaming". The emission mode gets stored in the programState as an Enum.

Closes #149

### CODE REVIEW

**Comment by rohrschacht** [comment]
Force-pushed to delete the hotfix commit which is already in master.

File: src/main/java/blf/configuration/EmissionSettings.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+public class EmissionSettings {
+
+    private static final Map<String, EmissionMode> emissionModeMap = new HashMap<>();
+
+    public static Map<String, EmissionMode> getEmissionModeMap() {
+        emissionModeMap.put("default batching", EmissionMode.DEFAULT_BATCHING);
+        emissionModeMap.put("save batching", EmissionMode.SAVE_BATCHING);
```
Comment (mykola-digtiar) [inline src/main/java/blf/configuration/EmissionSettings.java]: Save or safe? 🤔

Comment (Knostpe) [reply src/main/java/blf/configuration/EmissionSettings.java]: True was somehow convinced save batching was the right way to write it (e.g. savegame). Will change it just now!

**Review by daniil-romanchenko — DISMISSED** [review]
Looks fine, did you test it also by hand? 
I mean the new grammar functionality

**Review by daniil-romanchenko — COMMENTED** [review]
btw, what is exactly the difference between default batching, safe batching and streaming.
Does safe batching make files output after each block?

**Comment by Knostpe** [comment]
> Looks fine, did you test it also by hand?
> I mean the new grammar functionality

Yes tested if the information ends up in the programState in the right way using the new grammar (debugger). Edited: Wrong keywords have been tested aswell.

**Comment by Knostpe** [comment]
> btw, what is exactly the difference between default batching, safe batching and streaming.
> Does safe batching make files output after each block?

Safe batching refers to the "on the fly" functionality we have in another issue. If the program stops processing blocks for some reason before being at the last block, the program should be able to emit data which was extracted until this point with this mode. But of course this functionality still needs to be implemented :)

**Review by rohrschacht — DISMISSED** [review]
Looks good to me :+1:

**Review by mykola-digtiar — APPROVED** [review]
Amazing work! 👍

**Review by rohrschacht — APPROVED** [review]
Jenkins still passes after the merge :+1:

**Pull request merged**